### PR TITLE
chaterafrica define CSRF config

### PR DESCRIPTION
**Pull request by thepsalmist**
## Description

This build off #323 , allowing CORS enables requests from multiple domains as defined in `PAYLOAD_ALLOWED_DOMAINS`. However, payload's `login`, `logout` and `refresh` use HTTP cookies for auth, and doesn't allow authentication for any other domain (other than that defined during build time). These have to be defined in the `csrf` config.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Review by kilemensi — COMMENTED** [review]
👍🏽 

----

I understand your reasoning (especially how we use the domains so far) but I think these two settings should be configured via separate env vars.

Imagine we need to pull data from this site (say, news) to show on Code for Africa blog. We'd need to add `codeforafrica.org` domain to list of `CORS` domains but we 100% do not want to authenticate with that domain (`CSRF`).

File: apps/charterafrica/payload.config.ts [hunk 1]
```diff
@@ -33,7 +33,11 @@ dotenv.config({ path: path.resolve(__dirname, "./.env.local") });
 
 const appURL = process.env.PAYLOAD_PUBLIC_APP_URL;
 
-const allowedDomains = process?.env?.PAYLOAD_ALLOWED_DOMAINS?.split(",")
+const allowedCORSDomains = process?.env?.PAYLOAD_CORS_DOMAINS?.split(",")
+  ?.map((d) => d.trim())
+  ?.filter(Boolean);
+
+const allowedCSRFDomains = process?.env?.PAYLOAD_CSRF_DOMAINS?.split(",")
```
Comment (kilemensi) [inline apps/charterafrica/payload.config.ts:40]:
```suggestion
const cors = process?.env?.PAYLOAD_CORS?.split(",")
  ?.map((d) => d.trim())
  ?.filter(Boolean);

const csrf = process?.env?.PAYLOAD_CSRF?.split(",")
```

I take it these are just URLs and not necessary domains?

File: apps/charterafrica/payload.config.ts [hunk 2]
```diff
@@ -92,8 +96,8 @@ export default buildConfig({
       },
     }),
   },
-  cors: allowedDomains,
-  csrf: allowedDomains,
+  cors: allowedCORSDomains,
+  csrf: allowedCSRFDomains,
```
Comment (kilemensi) [inline apps/charterafrica/payload.config.ts:100]:
```suggestion
  cors,
  csrf,
```

**Review by kilemensi — APPROVED** [review]
🚀

File: apps/charterafrica/payload.config.ts (hunk 1 above)
Comment (thepsalmist) [reply apps/charterafrica/payload.config.ts:40]: Yes, these are URLS

**Review by kelvinkipruto — APPROVED** [review]

**Pull request merged**